alacritty_config_derive/src/serde_replace.rs

```diff
@@ -46,36 +46,41 @@ pub fn derive_recursive<T>(
         crate::generics_streams(&generics.params);
     let replace_arms = match_arms(&fields);
 
-    quote! {
-        #[allow(clippy::extra_unused_lifetimes)]
-        impl <'de, #constrained> alacritty_config::SerdeReplace for #ident <#unconstrained> {
-            fn replace(&mut self, value: toml::Value) -> Result<(), Box<dyn std::error::Error>> {
-                match value.as_table() {
-                    Some(table) => {
-                        for (field, next_value) in table {
-                            let next_value = next_value.clone();
-                            let value = value.clone();
-
-                            match field.as_str() {
-                                #replace_arms
-                                _ => {
-                                    let error = format!("Field \"{}\" does not exist", field);
-                                    return Err(error.into());
-                                },
-                            }
+    match replace_arms {
```

Comment:
This is going to be **far** cleaner if you early return on error. Just to deal with the already ridiculous amount of indentation.